### PR TITLE
fix: Rename JS Object failing in git child branch

### LINKED ISSUE #28766: [Bug]: Not able to rename JSObject in child branch
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

For git connected app, when we are on child branch, not able to rename the JSObject

### Steps To Reproduce

1. Connect an app to git, create jsObject
2. Create new branch, try renaming JSObject
3. Observe it doesnt rename

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
Renaming a JS object in child branch fails because the get action collections by pageId and branch name returns empty list. This PR fixes this problem.

#### PR fixes following issue(s)
Fixes #28766 #29131

#### Media
https://www.loom.com/share/ce94949a6272438389dc160e440b8c8c

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java [hunk 1]
```diff
@@ -114,9 +114,12 @@ public Flux<ActionCollection> findAllActionCollectionsByNamePageIdsViewModeAndBr
             }
 
             if (pageIds != null && !pageIds.isEmpty()) {
-                Criteria pageCriteria = where(fieldName(QActionCollection.actionCollection.publishedCollection) + "."
-                                + fieldName(QActionCollection.actionCollection.publishedCollection.pageId))
-                        .in(pageIds);
+                String pageIdFieldPath = String.format(
+                        "%s.%s.%s",
+                        fieldName(QActionCollection.actionCollection.publishedCollection),
+                        fieldName(QActionCollection.actionCollection.publishedCollection.defaultResources),
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java:120]: Please correct me if I am mistaken Nayan, but wouldn't this query return with action collections from all branches and not just the current one? I see it is being consumed in the get action collections by view mode method. For this method we would always expect to be working with the action collections in the current branch only.

Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java:120]: @nidhi-nair there is filter on the branch name as well. If you provide no branch name then it'll get actions from all branches. The code where it's consumed, isn't there any filter on branch?

Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java:120]: Ah I see. Does `branchName` inside defaultResources always point to current branch then? Can we please add a comment on the `DefaultResources` class indicating this? And rename this method as `findAllActionCollectionsByNameDefaultPageIdsViewModeAndBranch`? It's hard to remember, I defaulted to thinking this has to do with selecting "default branch" of the app.

**Comment by nidhi-nair** [comment]
Hey @nayan-rafiq, this will require changes in the EE repo as well. Let's raise a corresponding PR there as well so we can push both in together.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java (hunk 1 above)
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java:120]: @nidhi-nair I've updated the code. Also added comment to provide more clarity on the default resources properties.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**